### PR TITLE
Preparation for bootstrapping with UTXO set snapshot, part 2

### PR DESCRIPTION
**Pull request by kushti**
Another chunk of code extracted from #1444. In this PR:

* Unused entities `Application`,  `UTXOSnapshotChunk`, `UTXOSnapshotManifest` removed
* ModifierTypeId declarations scattered around the codebase now unified via `NetworkObjectTypeId` hierarchy

### CODE REVIEW

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 1]
```diff
@@ -1228,7 +1227,7 @@ object ErgoNodeViewSynchronizer {
     // getLocalSyncInfo messages
     case object SendLocalSyncInfo
 
-    case class ResponseFromLocal(source: ConnectedPeer, modifierTypeId: ModifierTypeId, localObjects: Seq[(ModifierId, Array[Byte])])
+    case class ResponseFromLocal(source: ConnectedPeer, modifierTypeId: NetworkObjectTypeId, localObjects: Seq[(ModifierId, Array[Byte])])
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:1230]: ScalaDoc missing

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 2]
```diff
@@ -1284,19 +1283,19 @@ object ErgoNodeViewSynchronizer {
       */
     case class FailedOnRecheckTransaction(id : ModifierId, error: Throwable) extends ModificationOutcome
 
-    case class RecoverableFailedModification(typeId: ModifierTypeId, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
+    case class RecoverableFailedModification(typeId: NetworkObjectTypeId.Value, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:1286]: ScalaDoc missing

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala [hunk 3]
```diff
@@ -1284,19 +1283,19 @@ object ErgoNodeViewSynchronizer {
...
 
-    case class SyntacticallyFailedModification(typeId: ModifierTypeId, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
+    case class SyntacticallyFailedModification(typeId: NetworkObjectTypeId.Value, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:1288]: ScalaDoc missing

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala [hunk 4]
```diff
@@ -117,7 +116,7 @@ trait ToDownloadProcessor extends BasicReaders with ScorexLogging {
     }
   }
 
-  def requiredModifiersForHeader(h: Header): Seq[(ModifierTypeId, ModifierId)] = {
+  def requiredModifiersForHeader(h: Header): Seq[(NetworkObjectTypeId.Value, ModifierId)] = {
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala:119]: ScalaDoc missing

File: src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala [hunk 5]
```diff
@@ -58,8 +55,8 @@ class HistoryStorage private(indexStore: LDBKVStore, objectsStore: LDBKVStore, c
     objectsStore.get(idToBytes(id)).map(_.tail) // removing modifier type byte with .tail
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala:55]: ScalaDoc missing here and for other methods

File: src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala [hunk 6]
```diff
@@ -717,12 +720,13 @@ object ErgoNodeViewHolder {
 
   case class BlockAppliedTransactions(txs: Seq[ModifierId]) extends NodeViewHolderEvent
 
-  case class DownloadRequest(modifiersToFetch: Map[ModifierTypeId, Seq[ModifierId]]) extends NodeViewHolderEvent
+  case class DownloadRequest(modifiersToFetch: Map[NetworkObjectTypeId.Value, Seq[ModifierId]]) extends NodeViewHolderEvent
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala:723]: ScalaDoc missing here and other classes

File: src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala [hunk 7]
```diff
@@ -1,20 +1,21 @@
...
-import scorex.core.{ModifierTypeId, NodeViewModifier}
+import scorex.core.NodeViewModifier
 import scorex.util.Extensions._
 import scorex.util.serialization.{Reader, Writer}
 import scorex.util.{ModifierId, ScorexLogging, bytesToId, idToBytes}
 
 import scala.collection.immutable
 
-case class ModifiersData(typeId: ModifierTypeId, modifiers: Map[ModifierId, Array[Byte]])
+case class ModifiersData(typeId: NetworkObjectTypeId.Value, modifiers: Map[ModifierId, Array[Byte]])
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala:16]: ScalaDoc missing, this and other classes

File: src/main/scala/scorex/core/network/DeliveryTracker.scala [hunk 8]
```diff
@@ -251,7 +253,7 @@ class DeliveryTracker(cacheSettings: NetworkCacheSettings,
       case _ => false
     }
 
-  def clearStatusForModifier(id: ModifierId, modifierTypeId: ModifierTypeId, oldStatus: ModifiersStatus): Unit =
+  def clearStatusForModifier(id: ModifierId, modifierTypeId: NetworkObjectTypeId.Value, oldStatus: ModifiersStatus): Unit =
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/network/DeliveryTracker.scala:256]: ScalaDoc missing

File: src/main/scala/scorex/core/network/DeliveryTracker.scala [hunk 9]
```diff
@@ -325,16 +327,16 @@ object DeliveryTracker {
   }
 
   case class FullInfo(
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/network/DeliveryTracker.scala:329]: ScalaDoc

File: src/main/scala/scorex/core/validation/ModifierError.scala [hunk 10]
```diff
@@ -1,19 +1,19 @@
 package scorex.core.validation
 
-import scorex.core.ModifierTypeId
+import org.ergoplatform.modifiers.NetworkObjectTypeId
 import scorex.util.ModifierId
 
 import scala.util.control.NoStackTrace
 
-case class InvalidModifier(error: String, modifierId: ModifierId, modifierTypeId: ModifierTypeId)
+case class InvalidModifier(error: String, modifierId: ModifierId, modifierTypeId: NetworkObjectTypeId.Value)
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/validation/ModifierError.scala:8]: ScalaDoc

File: src/main/scala/scorex/core/validation/ValidationSettings.scala [hunk 11]
```diff
@@ -11,7 +11,7 @@ import scorex.util.ModifierId
 abstract class ValidationSettings {
   val isFailFast: Boolean
 
-  def getError(id: Short, e: Throwable, modifierId: ModifierId, modifierTypeId: ModifierTypeId): Invalid =
+  def getError(id: Short, e: Throwable, modifierId: ModifierId, modifierTypeId: NetworkObjectTypeId.Value): Invalid =
     getError(id, InvalidModifier(e.getMessage, modifierId, modifierTypeId))
 
   def getError(id: Short, invalidMod: InvalidModifier): ValidationResult.Invalid
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/validation/ValidationSettings.scala:17]: ScalaDoc missing for all methods

File: src/main/scala/scorex/core/core.scala [hunk 12]
```diff
@@ -1,29 +1,25 @@
...
 import scorex.core.utils.ScorexEncoder
 import scorex.util.encode.Base16
 import supertagged.TaggedType
 
 package object core {
 
-  //TODO implement ModifierTypeId as a trait
-  object ModifierTypeId extends TaggedType[Byte]
-
   object VersionTag extends TaggedType[String]
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/core.scala:11]: ScalaDoc, what is VersionTag?

File: src/main/scala/scorex/core/NodeViewModifier.scala [hunk 13]
```diff
@@ -1,24 +1,24 @@
 package scorex.core
 
+import org.ergoplatform.modifiers.NetworkObjectTypeId
 import org.ergoplatform.modifiers.mempool.ErgoTransaction
 import scorex.core.serialization.BytesSerializable
 import scorex.core.utils.ScorexEncoding
 
-
 sealed trait NodeViewModifier extends BytesSerializable with ScorexEncoding {self =>
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/NodeViewModifier.scala:8]: ScalaDoc missing

File: src/main/scala/scorex/core/NodeViewModifier.scala [hunk 14]
```diff
@@ -1,24 +1,24 @@
...
   def id: scorex.util.ModifierId
 
-  def encodedId: String = encoder.encodeId(id)
-
   override def equals(obj: scala.Any): Boolean = obj match {
     case that: NodeViewModifier => (that.id == id) && (that.modifierTypeId == modifierTypeId)
     case _ => false
   }
 
+  def encodedId: String = id
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/NodeViewModifier.scala:20]: ScalaDoc missing

File: src/main/scala/scorex/core/NodeViewModifier.scala [hunk 15]
```diff
@@ -1,24 +1,24 @@
...
+
 }
 
 trait EphemerealNodeViewModifier extends NodeViewModifier
```
Comment (aslesarenko) [inline src/main/scala/scorex/core/NodeViewModifier.scala:24]: ScalaDoc missing

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: src/main/scala/scorex/core/NodeViewModifier.scala (hunk 15 above)
Comment (kushti) [reply src/main/scala/scorex/core/NodeViewModifier.scala:24]: For EphemerealNodeViewModifier ?? It is not touched!

File: src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala (hunk 5 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala:55]: done

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala (hunk 4 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala:119]: done

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala (hunk 1 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:1230]: Found no usages actually, removed the class

File: src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala (hunk 3 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala:1288]: done

File: src/main/scala/scorex/core/validation/ValidationSettings.scala (hunk 11 above)
Comment (kushti) [reply src/main/scala/scorex/core/validation/ValidationSettings.scala:17]: done (for the only method touched)

File: src/main/scala/scorex/core/NodeViewModifier.scala (hunk 14 above)
Comment (kushti) [reply src/main/scala/scorex/core/NodeViewModifier.scala:20]: done

File: src/main/scala/scorex/core/network/DeliveryTracker.scala (hunk 9 above)
Comment (kushti) [reply src/main/scala/scorex/core/network/DeliveryTracker.scala:329]: Not touched

File: src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala (hunk 6 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala:723]: Done (for this class touched only)

File: src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala (hunk 7 above)
Comment (kushti) [reply src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala:16]: Done (for this class touched only)

File: src/main/scala/scorex/core/validation/ModifierError.scala (hunk 10 above)
Comment (kushti) [reply src/main/scala/scorex/core/validation/ModifierError.scala:8]: Done

File: src/main/scala/scorex/core/network/DeliveryTracker.scala (hunk 8 above)
Comment (kushti) [reply src/main/scala/scorex/core/network/DeliveryTracker.scala:259]: Done

**Review by aslesarenko — APPROVED** [review]

**Pull request merged**